Use pkgconf to find libpostal

e-nomem opened the pull request:
`libpostal` is available on macOS via Homebrew now and it is one of the recommended ways to install the library, but homebrew on aarch64 installs the library into `/opt/homebrew/...` so the hard-coded `/usr/local/...` paths in the `setup.py` fail to find the installed library.

This PR instead calls out to `pkgconf` to determine the install paths. This works on macOS and linux (assuming that both `libpostal` and `pkgconf` are already installed and present on the system). I have not tested windows.

Notes:
 - This PR uses `subprocess.run()` which was introduced to the standard library in python 3.5, so I removed the trove classifiers for python 2

Fixes #92

e-nomem committed:
Use pkgconf to find libpostal

e-nomem committed:
Remove trove classifiers for python <3.5

albarrentine commented:
not sure I love introducing a system package dependency for the Python build, as there are a lot of Dockerfiles, various data platforms, etc. out there that it may break. Can either just add the /opt/homebrew paths to the list of include/lib paths, use an env var (looks like lxml does this https://github.com/lxml/lxml/blob/master/setup.py), or it looks like there’s a Python package on PyPI for pkgconf which I have not tested but if it works could maybe be installed as part of the build-system in a pyproject.toml (which we don’t have at present but is the standard nowadays).

e-nomem commented:
Any of those sound good to me and I'm happy to implement it. Do you have a strong preference? If not I'll probably go the route of adding a `pyproject.toml`

e-nomem committed:
Check env vars and fall back to defaults

e-nomem commented:
Turns out the python pkgconf library also just shells out to the binary like my current code does.

For flexibility I've added a `LIBPOSTAL_PREFIX` env var that is checked first. Modern tooling like `uv` and `build` enforce isolation pretty well though so this is hard to pass down. It only really works for the `python setup.py ...` invocation.

For compatibility, I've also added a fallback to the existing `/usr/local/{include,lib}` paths if neither the env var nor pkgconf work out.

e-nomem committed:
Handle case when libpostal.pc is not found

albarrentine committed:
[fix] simplifying setup.py
